mdal/mdal_memory_data_model.hpp

```diff
@@ -34,29 +36,47 @@ namespace MDAL
   class MemoryDataset2D: public Dataset2D
   {
     public:
-      MemoryDataset2D( DatasetGroup *grp );
+      MemoryDataset2D( DatasetGroup *grp, bool hasActiveFlag = false );
       ~MemoryDataset2D() override;
 
       size_t scalarData( size_t indexStart, size_t count, double *buffer ) override;
       size_t vectorData( size_t indexStart, size_t count, double *buffer ) override;
+
+      //! Returns 0 for datasets that does not support active flags
       size_t activeData( size_t indexStart, size_t count, int *buffer ) override;
 
       /**
-       * valid pointer only for for dataset defined on vertices
+       * Loop through all faces and activate those which has all 4 values on vertices valid
+       * Dataset must support active flags and be defined on vertices
        */
-      int *active();
-      double *values();
+      void activateFaces( MDAL::MemoryMesh *mesh );
 
-      const int *constActive() const;
-      const double *constValues() const;
+      /**
+       * Sets active flag for index
+       *
+       * \param stat 1 for active, 0 for non-active
+       * \param index index of the flag
+       *
+       * Dataset must support active flags
+       */
+      void setActive( size_t index, int stat );
+      void setActive( const int *activeBuffer );
+      int active( size_t index ) const;
+
+
+      void setValue( size_t index, double value );
+      void setValue( size_t index, double x, double y );
```

Comment:
It would be good to put implementation of these methods to the header file so they can be inlined by the compiler... (the bodies are very simple anyway)